src/util.js

```diff
@@ -358,7 +358,12 @@ function lineShouldEndWithSemicolon(path) {
 }
 
 function lineShouldHaveEndPHPTag(path) {
-  return isNextNodeInline(path);
+  const node = path.getValue();
+  const isProgramNode = node.kind === "program";
+  return (
+    isNextNodeInline(path) ||
+    (isProgramNode && node.loc.source.trim().endsWith("?>"))
```

Comment:
Looks good, I'm not sure that this check is good, but we can add more tests in future and fix this problems in future